Framework/Algorithms/src/ChangePulsetime2.cpp

```diff
@@ -0,0 +1,63 @@
+#include "MantidAlgorithms/ChangePulsetime2.h"
+#include "MantidAPI/WorkspaceFactory.h"
+#include "MantidDataObjects/EventWorkspace.h"
+#include "MantidKernel/ArrayProperty.h"
+#include "MantidKernel/System.h"
+
+namespace Mantid {
+namespace Algorithms {
+
+// Register the algorithm into the AlgorithmFactory
+DECLARE_ALGORITHM(ChangePulsetime2)
```

Comment:
It would be nice to make a proper v2 of this algorithm and distribute it as part of these changes so that it's clear for future developers how these features are to be used. It does however yield a breaking change to the property names. In it's current form, I think this algorithm must be omitted from our public/user algorithm namespaces to prevent real use and dependency on.